app/controllers/api/posts_controller.rb

```diff
@@ -1,8 +1,30 @@
 class Api::PostsController < Api::ApiController
-  before_action :authenticate_api_user!, only: [ :create ]
+  respond_to :json
+
   def index
-    respond_to do |format|
-      format.json
+    @posts = Post.with_attached_image.order("created_at DESC")
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.user = current_user
+    @post.image.attach(io: image_io, filename: "image")
+
+    if @post.save
```

Comment:
return the post json?

Reply:
Does this look okay?

Reply:
ohhh, hah. Because jbuilder. I think you want to turn this to `unless @post.save` to remove the else